low-level/files/opt/seagate/sspl/setup/resource_map/storage.py

```diff
@@ -128,6 +136,74 @@ def get_controllers_info(self):
             data.append(controller_dict)
         return data
 
+    def get_logical_volumes_info(self):
+        """Update and return logical volume information in specific format."""
+        logvol_data = []
+        logicalvolumes = self.get_realstor_encl_data("volumes")
+        if logicalvolumes:
+            for logicalvolume in logicalvolumes:
+                health = logicalvolume.get("health", "NA")
+                pool_sr_no = logicalvolume.get("container-serial", "NA")
+                uid = logicalvolume.get("volume-name", "NA")
+                if pool_sr_no in self.dg_vol_map:
```

Comment:
Can you explain logic from line 148-151, whats the intent of having this dg_vol_map[] 

If pool_number  found, then append ! why? update needed here?
if not found, update !? Shouldn't be updating in this case ?

Reply:
1) if pool_sr_no in self.dg_vol_map:
2)      self.dg_vol_map[pool_sr_no].append({"volume_uid": uid})
3) else:
4)      self.dg_vol_map.update({pool_sr_no:[{"volume_uid": uid}]})

On the above code, we are adding volumes that are related to a particular poll number (disk group).
1st line we will check if the pool number is already present in the dictionary, then we will add only volume uid under that pool number, otherwise in line no 4 we are creating a new pool entry (key) in the dictionary and starting to add volume uid which is related to that pool.

Note: **self.dg_vol_map** is required in disk group info.

Reply:
Ok, i think i understand now. First pool number should exist in dg_vol_map[] to add volume_uid.